### ISSUE TITLE
ComboBox.getValue() returns null after adding an item

### ISSUE THREAD

**Issue by pinonpierre** [vaadin/framework#11385]
Hi,

ComboBox still have some issues...

Code Snippet:

```
        VerticalLayout mainLayout = new VerticalLayout();
        setCompositionRoot(mainLayout);

        List<String> items = new ArrayList<>(Arrays.asList("blue", "red", "green"));
        ListDataProvider<String> dataProvider = new ListDataProvider<>(items);

        ComboBox<String> comboBox = new ComboBox<>();
        comboBox.setPageLength(0);
        comboBox.setDataProvider(dataProvider);
        comboBox.setNewItemProvider(value -> {
            items.add(value);
            dataProvider.refreshAll();
            return Optional.ofNullable(value);
        });
        mainLayout.addComponent(comboBox);

        Button getValueButton = new Button("Get Value");
        getValueButton.addClickListener(event1 -> event1.getButton().setCaption("Get Value: [" + comboBox.getValue() + "]"));
        mainLayout.addComponent(getValueButton) ;
```

Steps to reproduce:
- Enter new Value like "aaa" in the ComboBox
- Click on the button
- The value returned is "null" !

Demo:
![combobox_bug](https://user-images.githubusercontent.com/4507910/50098759-dbb62d00-021c-11e9-9a43-ad6a52098e62.gif)

**Comment by pinonpierre** [comment]
Maybe a link with these issues??

https://github.com/vaadin/framework/issues/11317
https://github.com/vaadin/framework/issues/11320

**Comment by pinonpierre** [comment]
Hi,

For information, if it can help you to fix the issue, I find a workaround:

Replacing comboBox.setPageLength(0) by comboBox.setPageLength(Integer.MAX_VALUE) seems to fix the issue.

I hope, I will not have side effect...

Pierre,

**Comment by pinonpierre** [comment]
Seems to be fixed in 8.8.0
